pkg/querier/querier_test.go

```diff
@@ -132,14 +154,120 @@ var (
 	}
 )
 
+func TestShouldSortSeriesIfQueryingMultipleQueryables(t *testing.T) {
+	start := time.Now().Add(-2 * time.Hour)
+	end := time.Now()
+	ctx := user.InjectOrgID(context.Background(), "0")
+	var cfg Config
+	flagext.DefaultValues(&cfg)
+	overrides, err := validation.NewOverrides(DefaultLimitsConfig(), nil)
+	const chunks = 1
+	require.NoError(t, err)
+
+	labelsSets := []labels.Labels{
+		{
+			{Name: model.MetricNameLabel, Value: "foo"},
+			{Name: "order", Value: "1"},
+		},
+		{
+			{Name: model.MetricNameLabel, Value: "foo"},
+			{Name: "order", Value: "2"},
+		},
+	}
+
+	db, samples := mockTSDB(t, labelsSets, model.Time(start.Unix()*1000), int(chunks*samplesPerChunk), sampleRate, chunkOffset, int(samplesPerChunk))
+	distributor := &MockDistributor{}
+
+	unorderedResponse := client.QueryStreamResponse{
+		Timeseries: []cortexpb.TimeSeries{
+			{
+				Labels: []cortexpb.LabelAdapter{
+					{Name: model.MetricNameLabel, Value: "foo"},
+					{Name: "order", Value: "2"},
+				},
+				Samples: samples,
+			},
+			{
+				Labels: []cortexpb.LabelAdapter{
+					{Name: model.MetricNameLabel, Value: "foo"},
+					{Name: "order", Value: "1"},
+				},
+				Samples: samples,
+			},
+		},
+	}
+	distributor.On("QueryStream", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(&unorderedResponse, nil)
+
+	tCases := []struct {
+		name                 string
+		distributorQueryable QueryableWithFilter
+		queriables           []QueryableWithFilter
+		sorted               bool
+	}{
+		{
+			name:                 "should sort if querying 2 queryables",
+			distributorQueryable: newDistributorQueryable(distributor, cfg.IngesterStreaming, cfg.IngesterMetadataStreaming, batch.NewChunkMergeIterator, cfg.QueryIngestersWithin),
```

Comment:
A small nit:
The `newDistributorQueryable` function is a bit long. Can we move it at line 200 use that distributor queryable?

Reply:
done